src/main/pages/che-7/installation-guide/assembly_airgap-installation.adoc

```diff
@@ -0,0 +1,167 @@
+====
+
+The main purpose of building offline registry images is to ensure that workspaces can be started without relying on resources from the outside internet. For the plugin registry, this means building a registry that serves all plugin or extension artifacts in addition to plugin metadata. Similarly, an offline devfile registry serves all sample projects referenced in devfiles as `zip` files.
+
+The plugin registry repositories contain scripts to make it easier to build custom images.
+
+
+### Offline plugin registry
+[source,bash]
+----
+git clone https://github.com/eclipse/che-plugin-registry.git <1>
+cd che-plugin-registry
+git checkout 7.3.0 // <2>
+# Build the image
+./build.sh --organization <my-org> \
+           --registry <my-registry> \
+           --tag <my-tag> \
+           --offline // <3>
+----
+<1> Clone the plugin registry repository
+<2> Check out version of registry to be deployed
+<3> Full options for the `build.sh` script can be found by passing parameter `--help`
+
+This build of the registry will build a docker image that contains all `.vsix` and `.theia` extensions that can be used in workspaces; since this results in a quite large image, a version of the registry that contains only the _latest_ version of each plugin can be built using the argument `--latest-only`.
+
+
+### Offline devfile registry
+[source,bash]
+----
+git clone https://github.com/eclipse/che-devfile-registry.git <1>
+cd che-devfile-registry
+git checkout 7.3.0 // <2>
+# Build the image
+./build.sh --organization <my-org> \
+           --registry <my-registry> \
+           --tag <my-tag> \
+           --offline // <3>
+----
+<1> Clone the devfile registry repository
+<2> Check out version of registry to be deployed
+<3> Full options for the `build.sh` script can be found by passing parameter `--help`
+
+
+## Pulling images used by Che to a private registry
+As Che workspaces consist of pods running on a cluster, a fairly large number of images are required to support all languages and plugins that could be used for a workspace. In an air-gap or offline deployment, these images must be made available to the Che server running in the cluster.
+
+### Essential images
+In general, the images _essential_ to starting a workspace are infrastructure images those that are included in every workspace launch:
+
+[cols="2*"]
+|===
+| `docker.io/eclipse/che-server`
+| The main Che server image
+
+| `docker.io/eclipse/che-postgres`
+| The database used by Che
+
+| `docker.io/eclipse/che-keycloak`
+| Keycloak pod used for user authentication
+
+| `quay.io/eclipse/che-jwtproxy`
+| JWT proxy image used for enabling authentication between services. See: link:{site-baseurl}pages/che-7/overview/con_che-workspace-jwt-proxy.html[additional documentation]
+
```

Comment:
It appears you're missing che-theia, che-machine-exec, and che-theia-runtime-endpoint-binary images.

Reply:
Those are just plugin images, they're covered by the instructions for plugin registry.

Reply:
E.g. for prod-preview: https://che-plugin-registry.prod-preview.openshift.io/v3/external_images.txt

Reply:
PR for making sure the remote runtime container is included in this: https://github.com/eclipse/che-plugin-registry/pull/282